### PR TITLE
GCR reader for DC2 static coadd catalogs

### PR DESCRIPTION
**Pull request by yymao**
This PR adds a GCR reader for the DC2 static coadd catalogs generated by @wmwv in #153 and also [an example notebook](https://github.com/LSSTDESC/DC2_Repo/blob/u/yymao/GCRforDC2Coadd/Notebooks/DC2%20Static%20Coadd%20Catalog%20reader.ipynb) of how to use it.

### CODE REVIEW

**Comment by wmwv** [comment]
The LSST Project refers to its 'y' filter as lower-case, and I presume this will be the case for distributed data products as well.

**Comment by RobertLuptonTheGood** [comment]
HSC also calls it y.   Y is usually reserved for the HgCdTe c. 1 micron band.

**Comment by wmwv** [comment]
Expand patches back out as strings.

In the LSST current skymap scheme patches are strings.  E.g., '0,4', 1,2' or '2,3'.  It is often convenient to represent them as '04', '12', or '23', but those shouldn't be translated back into ints of 4, 12, 23.  Such ints are of course unique, but it's less obvious how to get all of the patches in a given row or column and makes the neighbor relationship less clear.

**Comment by yymao** [comment]
thanks for the feedback!

File: scripts/reader/dc2_static_coadd_reader.py [hunk 1]
```diff
@@ -10,36 +10,45 @@
 
 class DC2StaticCoaddCatalog(BaseGenericCatalog):
 
-    _native_filter_quantities = {'tract', 'patch'}
+    _native_filter_quantities = {'tract', 'patch_ra', 'patch_dec'}
```
Comment (wmwv) [inline scripts/reader/dc2_static_coadd_reader.py]: I don't like patch_ra and patch_dec, although I'm fine with dividing them up.  There's nothing that says that patches have to be oriented along RA or Dec.

Comment (wmwv) [reply scripts/reader/dc2_static_coadd_reader.py]: Do you need an int lookup for something?  Is there a reason you can't just have
```
_native_filter_quantities = {'tract', 'patch'}
```
but have patch be a str?

Comment (yymao) [reply scripts/reader/dc2_static_coadd_reader.py]: @wmwv it surely can. I've updated per your suggestions. The only downside is that people cannot do `(lambda p: p < 2, 'patch_ra')` as a native filter, but I agree with you that this is probably not a common use case.

**Comment by wmwv** [comment]
@yymao Could you merge this in in advance of the DESC Seminar tomorrow (May 30)?

**Comment by yymao** [comment]
Sure. If there's no objection I'll merge this tonight. 

(cc @jchiang87 @katrinheitmann @cwwalter)

**Pull request merged**